src/module.ts

```diff
@@ -1,5 +1,12 @@
 /* eslint-disable @typescript-eslint/ban-ts-comment */
-import { defineNuxtModule, addVitePlugin, extendWebpackConfig } from '@nuxt/kit'
+import { fileURLToPath } from 'node:url'
+import {
+  defineNuxtModule,
+  addVitePlugin,
+  extendWebpackConfig,
+  createResolver,
+  addComponent
```

Comment:
I had a misconception about `addComponent`, I thought it could make trouble for webpack users by eager loading it
but I guess it's not that way, so I changed it back to the way it does it in nuxt-icons in a commit now